src/metaswitch/ellis/remote/homestead.py

```diff
@@ -259,14 +259,19 @@ def callback_wrapper(response):
             _log.debug("503 response - retrying %d more time(s)..." % retries_holder['retries'])
             retries_holder['retries'] -= 1
             # Set a timer to retry the HTTP request in 500ms.
-            IOLoop.instance().add_timeout(datetime.timedelta(milliseconds=500), do_http_request)
+            if retries_holder['retries'] >= 3:
+                IOLoop.instance().add_timeout(datetime.timedelta(seconds=1), do_http_request)
```

Comment:
Aren't these the wrong way round? Surely the timeout should be longer on the later retries?

Reply:
I think it's right. `retries_holder['retries']` holds the number of retries we have left, so later retires have a lower value of `retries_holder['retries']`?